SOD.Common/Custom/InputSuppressionEntry.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections;
+using BepInEx;
+using Rewired;
+using UnityEngine;
+using UniverseLib;
+
+namespace SOD.Common.Custom
+{
+    public sealed class InputSuppressionEntry
+    {
+        public InputSuppressionEntry(string callerGuid, KeyCode keyCode, TimeSpan? time = null)
+        {
+            CallerGuid = callerGuid;
+            KeyCode = keyCode;
+            InteractionKey = InteractablePreset.InteractionKey.none;
+            ElementIdentifierName = string.Empty;
+            ElementIdentifierId = -1;
+            TimeRemainingSec = (float)(time?.TotalSeconds ?? 0d);
+            IsSetOnInteractionKey = false;
+        }
+
+        public InputSuppressionEntry(string callerGuid, InteractablePreset.InteractionKey interactionKey, TimeSpan? time = null)
+        {
+            CallerGuid = callerGuid;
+            InteractionKey = interactionKey;
+            var binding = Lib.InputDetection.GetBinding(interactionKey);
+            ElementIdentifierName = binding.elementIdentifierName;
+            ElementIdentifierId = binding.elementIdentifierId;
+            KeyCode = Lib.InputDetection.GetApproximateKeyCode(binding);
+            TimeRemainingSec = (float)(time?.TotalSeconds ?? 0d);
+            IsSetOnInteractionKey = true;
+        }
+
+        public KeyCode KeyCode { get; }
+        public string ElementIdentifierName { get; }
+        public int ElementIdentifierId { get; }
+        public string CallerGuid { get; }
+        public InteractablePreset.InteractionKey InteractionKey { get; }
+        public bool IsSetOnInteractionKey { get; }
+        public float TimeRemainingSec { get; set; }
+
+        private Coroutine _coroutine;
+
+        public void Start()
+        {
+            if (_coroutine != null || TimeRemainingSec <= 0f)
+                return;
+            _coroutine = RuntimeHelper.StartCoroutine(Tick());
+        }
+
+        public void Stop()
+        {
+            if (_coroutine == null) return;
+            RuntimeHelper.StopCoroutine(_coroutine);
+            _coroutine = null;
+        }
+
+        internal string ConvertToDictionaryKey()
+        {
+            return Lib.InputDetection.ConvertToDictionaryKey(CallerGuid, InteractionKey, KeyCode);
```

Comment:
Cache this in constructor, it can't change anymore right?

Reply:
yeah, you're right, I'll change it